gpt_index/dataset_generation/__init__.py

```diff
@@ -0,0 +1,5 @@
+"""Dataset generation module."""
```

Comment:
instead of creating a new folder here, can you add the dataset generation module as part of `evaluation` folder? You can name the file `dataset_generation.py`.

And add `DatasetGenerator` to `evaluation/__init__.py`.